src/main/java/at/hannibal2/skyhanni/features/rift/CruxTalismanDisplay.kt

```diff
@@ -0,0 +1,120 @@
+package at.hannibal2.skyhanni.features.rift
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.getInternalName
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils.addAsSingletonList
+import at.hannibal2.skyhanni.utils.NumberUtil.roundToPrecision
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStringsAndItems
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class CruxTalismanDisplay {
+
+    private val config get() = SkyHanniMod.feature.rift.cruxTalisman
+    private val partialName = "CRUX_TALISMAN"
+    private var display = listOf<List<Any>>()
+    private val displayLine = mutableListOf<Crux>()
+    private val bonusesLine = mutableListOf<String>()
+    private val progressPattern = ".*(?<tier>§[0-9a-z][IV1-4-]+)\\s+(?<name>§[0-9a-z]\\w+)§[0-9a-z]:\\s*(?<progress>§[0-9a-z](?:§[0-9a-z])?MAXED|(?:§[0-9a-z]\\d+§[0-9a-z]\\/§[0-9a-z]\\d+)).*".toPattern()
+    private var maxed = false
+    private var percent = 0.0
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GameOverlayRenderEvent) {
+        if (!isEnabled()) return
+        config.position.renderStringsAndItems(
+                display,
+                posLabel = "Crux Talisman Display"
+        )
+    }
+
+    private fun update() {
+        display = drawDisplay()
+    }
+
+    private fun drawDisplay(): List<List<Any>> {
+        return buildList {
+            var i = 0
+            var p = 0
+            for (crux in displayLine)
+                if (crux.maxed)
+                    i++
+            if (i == 6)
+                maxed = true
+
+            if(!config.compactWhenMaxed && maxed) maxed = false
+
+            if (displayLine.isNotEmpty()) {
+                addAsSingletonList("§7Crux Talisman Progress: ${if (maxed) "§a§lMAXED!" else "§a$percent%"}")
+                if (!maxed) {
+                    displayLine.forEach {
+                        if (config.compactWhenMaxed) {
+                            if (!it.maxed) {
+                                p += it.progress.removeColor().split("/")[0].toInt()
+                                addAsSingletonList("  ${it.tier} ${it.name}: ${it.progress}")
+                            } else {
+                                addAsSingletonList("  ${it.tier} ${it.name}: ${it.progress}")
+                                p += 100
+                            }
+                        } else {
+                            addAsSingletonList("  ${it.tier} ${it.name}: ${it.progress}")
```

Comment:
These 3 addAsSingletonList calls can be moved outside of the 2 if blocks, right?

Reply:
Since they look identical to me